### DISCUSSION TITLE
How to render nested blocks in a repeater?

### DISCUSSION THREAD

**Discussion by dweerd** [area17/twill#2269] [Q&A]
I have an accordion block, which has a repeater for accordion items. Every accordion item has a block editor, to allow nested blocks.

Normally to render nested blocks you can say something like `{!! $renderData->renderChildren('subblocks') !!}`, but how to render the subblocks of each of the repeater items?

```
@foreach ($block->children()->orderBy('position')->get() as $child)
    {{--  How to render this childs' subblocks? --}}
@endforeach
```

**Comment by plokko** [comment]
Hi, i didn't find a solution so i did a bit of reverse enginnering and i found a solution, it could be prettier but it works.

Let's say you have a block with a repeater and on each repeater item you have a 'class' input and a block called 'content' (it was a customizable grid) 

This is the code of your block view (*resources/views/site/blocks/<yourblockname>.blade.php*):
```php
<div class="row">
    {{-- - You must iterate over $renderData and NOT $block childrens --}}
    @foreach ($renderData->children as $childRender)
        @php
            $childRenderData = $childRender->renderData; ///<<< this is the renderData of the child
            $child = $childRenderData->block; ///<<< this is our child block
        @endphp
        <div class="{{ $child->input('class') ?? 'col' }}">{!! $childRenderData->renderChildren('content') !!}</div>
    @endforeach
</div>
```

**Reply by oreoluwa-bs** [reply]
Thank you
